### DISCUSSION TITLE
Why streaming process needn't call `reset_states()` after model process?

### DISCUSSION THREAD

**Discussion by MLrookie** [snakers4/silero-vad#206] [Q&A] [answered]
thank you for your excellent work!
i have something confused about the streaming process of silero-vad model. In streaming process example, the model process a chunk audio of 250ms every time, but it don't use `reset_states()` after each process. So, what is the function of `reset_states()`, can i use silero-vad model without using `reset_states()` continuously?

**Comment by snakers4** [comment, marked as answer]
Hi,

> In streaming process example, the model process a chunk audio of 250ms every time, but it don't use reset_states() after each process.

```
## using VADIterator class

vad_iterator = VADIterator(model)
wav = read_audio(f'en_example.wav', sampling_rate=SAMPLING_RATE)

window_size_samples = 1536 # number of samples in a single audio chunk
for i in range(0, len(wav), window_size_samples):
    chunk = wav[i: i+ window_size_samples]
    if len(chunk) < window_size_samples:
      break
    speech_dict = vad_iterator(chunk, return_seconds=True)
    if speech_dict:
        print(speech_dict, end=' ')
vad_iterator.reset_states() # reset model states after each audio
```

I believe that the [example](https://github.com/snakers4/silero-vad/wiki/Examples-and-Dependencies#examples) has about 96ms, not 250ms. Maybe you were using an old version, which was [deprecated](https://github.com/snakers4/silero-vad/issues/2#issuecomment-987878454).

> but it don't use reset_states() after each process.

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L380-L387

It does reset state BEFORE streaming.
It is assumed that after the stream ends, the process is killed.
For a new stream the state is reset again, etc.

>  So, what is the function of reset_states(), can i use silero-vad model without using reset_states() continuously?

The VAD keeps some internal state.
You can use it continuously without resettting the state provided this is the same stream of audio.
If you start another audio stream, you need to reset the states.
This thread may be helpful - https://github.com/snakers4/silero-vad/discussions/142/
